core/fetcher/fetcher.go

```diff
@@ -127,11 +127,21 @@ func (f *Fetcher) fetchAttesterData(ctx context.Context, slot int64, defSet core
 
 		eth2AttData, ok := dataByCommIdx[commIdx]
 		if !ok {
-			var err error
-			eth2AttData, err = f.eth2Cl.AttestationData(ctx, eth2p0.Slot(uint64(slot)), commIdx)
+			var (
+				err      error
+				eth2Resp *eth2api.Response[*eth2p0.AttestationData]
```

Comment:
any reason behind defining `eth2Resp` here before calling the endpoint?
we can do what we are already doing on line 218 in the same file.

Reply:
yeah, modified